arachne_development/server/CCMsg.chpl

```diff
@@ -86,714 +84,846 @@ module CCMsg {
     var gEntry:borrowed GraphSymEntry = getGraphSymEntry(graphEntryName, st);
     var ag = gEntry.graph;
 
-    // Implementation of the algorithm for undirected graphs, they can be 
-    // weighted or unweighted. 
-    proc cc_kernel_und(nei:[?D1] int, start_i:[?D2] int,src:[?D3] int, dst:[?D4] int, neiR:[?D11] int, start_iR:[?D12] int,srcR:[?D13] int, dstR:[?D14] int):string throws { 
-      // Look for the first instance of -1 and get the first vertex to
-      // start BFS at. The first component is obviously component 1. 
 
-      // Change the visited array to all -1. 
-      coforall loc in Locales {
-        on loc {
-          forall i in visited.localSubdomain() {
-            visited[i] = -1; 
-          }
-        }
+
+
+    // Third implemention of the fast shiloach-vishkin algorithm for connected components proposed 
+    // by Yongzhe Zhang, Ariful Azad, and Zhenjiang Hu.
+    proc cc_fast_sv(nei:[?D1] int, start_i:[?D2] int,src:[?D3] int, dst:[?D4] int, neiR:[?D11] int, start_iR:[?D12] int,srcR:[?D13] int, dstR:[?D14] int) throws {
+      // Initialize the parent vectors f that will form stars. 
+      var f = makeDistArray(Nv, int); 
+      var f_low = makeDistArray(Nv, int); 
+      var localtimer:stopwatch;
+      var myefficiency:real;
+      var executime:real;
+
+      forall i in 0..Nv-1 {
```

Comment:
This `forall` loop actually forces the entire creation of the values for `i` on `locale0` and writes them to the locale that owns `f[i]`. To make this trully multilocale you have to do something like

```chapel
for i in f.domain {
     f[i] = i;
     f_low[i] = i;
}
Since the `domain` of `f` is distributed, then the `forall` loop will create a multilocale iterator to do the assignment at each locale.

Reply:
I am not sure what code you mentioned. If you mean the forall loop like this.
https://github.com/Bears-R-Us/arkouda-njit/blob/multilocale/arachne_development/server/CCMsg.chpl#L827-L829
I cannot use a distributed way here because I want to have a local array that contains all the values of the distributed array.
localf is an array that only exists in the current locale. However, its size is the same as the distributed array af. This is the reason I cannot use the way like
localf.localSubdomain()
to execute the assignments on different locales.